src/test/ui/attributes/register-attr-tool-fail.rs

```diff
@@ -0,0 +1,13 @@
+#![feature(register_attr)]
+#![feature(register_tool)]
+
+#![register_attr] //~ ERROR malformed `register_attr` attribute input
+#![register_tool] //~ ERROR malformed `register_tool` attribute input
+
+#![register_attr(a::b)] //~ ERROR `register_attr` only accepts identifiers
+#![register_tool(a::b)] //~ ERROR `register_tool` only accepts identifiers
+
+#![register_attr(attr, attr)] //~ ERROR attribute `attr` was already registered
+#![register_tool(tool, tool)] //~ ERROR tool `tool` was already registered
+
```

Comment:
what's the result of attr vs tool collision?

```rust
#![register_attr(something)]
#![register_tool(something)]
```

Reply:
They do not collide because the attribute is in macro namespace and the tool is in type namespace.